client/src/components/rooms/RoomsContainer.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from "react";
+import { useGetRoom } from "../../hooks/useGetRoom";
+import { scrollToTop } from "../../hooks/useScrollToTop";
+import { Link } from "react-router-dom";
+import { useAuthContext } from "../../context/authContext";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+export const RoomsContainer = () => {
+  const { authUser } = useAuthContext();
+  const { rooms, updateRoomHandler } = useGetRoom();
+
+  const [updatedName, setUpdatedName] = useState("");
+  const [updatedAbout, setUpdatedAbout] = useState("");
+  const [updatedPrice, setUpdatedPrice] = useState("");
+  const [updatedInfo, setUpdatedInfo] = useState("");
+  const [selectedRoom, setSelectedRoom] = useState(null);
+  // eslint-disable-next-line
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  const handleUpdate = (item) => {
+    setSelectedRoom(item);
+    setUpdatedName(item.name);
+    setUpdatedAbout(item.about);
+    setUpdatedPrice(item.price);
+    setUpdatedInfo(item.info);
+  };
+
+  const handleSaveUpdate = async (id) => {
+    try {
+      await updateRoomHandler(
+        id,
+        updatedName,
+        updatedAbout,
+        updatedPrice,
+        updatedInfo,
+      );
+      setSelectedRoom(null);
+      setIsUpdating(false);
+    } catch (error) {
+      toast.error(error.message);
+    }
+  };
+
+  return (
+    <div className="rooms-container">
+      {rooms && rooms.length > 0 ? (
+        rooms.map((room) => (
+          <div key={room._id} className="room">
```

Comment:
Same moment as in Reservation - ReservationItem. I suggest moving into a new component RoomItem and then just map it here to simplify the map method

Reply:
Done